copier/renderer.py

```diff
@@ -0,0 +1,174 @@
+from collections import ChainMap
+from dataclasses import dataclass, field
```

Comment:
Probably you should change this to a pydantic dataclass, just to be consistent with the rest of Copier.